common/security.h

```diff
@@ -32,6 +32,30 @@ inline int hasUID(const char *userId)
     return 0;
 }
 
+inline int isInContainer()
+{
+#ifdef __linux__
+    FILE *cgroup;
+    char line[80];
+    const char *docker = "/docker/";
+    cgroup = fopen("/proc/self/cgroup", "r");
+    if(!cgroup)
+    {
+        fprintf(stderr, "Error: cannot open /proc/self/cgroup\n");
+        return 0;
+    }
+    if (fgets(line, sizeof(line), cgroup) == NULL)
+    {
+        fprintf(stderr, "Error: cannot read from /proc/self/cgroup\n");
+        return 0;
+    }
+    fclose(cgroup);
```

Comment:
I guess we should line[sizeof(line)-1] = 0; here before the strstr to be safe?

Reply:
fgets() inserts the terminating \0

Reply:
Ah you are right, thanks :-)